src/Server/src/GraphQL/Client/ClientQueries.cs

```diff
@@ -37,13 +38,34 @@ public async Task<Client> GetClientAsync(Guid id, CancellationToken cancellation
         }
 
         [AuthorizeClientAuthoring(AccessMode.Read, includeTenantAuth: true)]
-        public async Task<SearchResult<Client>> SearchClientsAsync(
-            SearchClientsRequest input,
+        public async Task<SearchResult<Client>> SearchClientsAsync(SearchClientsRequest input,
             CancellationToken cancellationToken)
         {
-            return await _clientService.SearchClientsAsync(
-                input,
-                cancellationToken);
+            return await _clientService.SearchClientsAsync(input, cancellationToken);
+        }
+
+        [AuthorizeClientAuthoring(AccessMode.Read, includeTenantAuth: false)]
+        public async Task<GetClientSecretPayload> GetClientSecretAsync(GetClientSecretRequest input,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                (Client client, string secret) =
+                    await _clientService.GetClientSecretAsync(input, cancellationToken);
+                return new GetClientSecretPayload(client, secret);
+            }
+            catch (ErrorException e)
+            {
+                Client client =
+                    await _clientService.GetByIdAsync(input.ClientId, cancellationToken);
+
+                if (e.Error is NoEncryptedSecretError error)
```

Comment:
Here you have to ask yourself, is it really an error case?
Or is the exception used here as control flow?

If it is expected that there can be clients with no secret, an exception would be wrong